packages/teleterm/src/ui/TopBar/Connections/ConnectionsFilterableList/ConnectionItem.tsx

```diff
@@ -48,37 +48,43 @@ export function ConnectionItem(props: ConnectionItemProps) {
         height: unset;
       `}
     >
-      <ConnectionStatusIndicator mr={3} connected={props.item.connected} />
+      <ConnectionStatusIndicator
+        mr={3}
+        css={`
+          flex-shrink: 0;
+        `}
+        connected={props.item.connected}
+      />
       <Flex
         alignItems="center"
         justifyContent="space-between"
         flex="1"
-        width="100%"
         minWidth="0"
       >
-        <Flex flexDirection="column">
+        <div
+          css={`
+            min-width: 0;
+          `}
+        >
```

Comment:
There's [a really nice answer on SO](https://stackoverflow.com/a/43809765/742872) about it. Flex children by default have `min-width` set to `auto`, so `text-overflow` will never work on them.

Instead, we have to, in a cascading fashion, set `min-width` on those flex children which have content that should get hidden on overflow. So we have `min-width` on this div here and one above it.